File: engine-common/src/test/java/com/sos/scheduler/engine/common/LazyTest.java
package com.sos.scheduler.engine.common;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.MatcherAssert.assertThat;

public final class LazyTest {
    private int i = 0;

    @Test public void testApply() throws Exception {
        Lazy<Integer> lazy = new Lazy<Integer>() {
            @Override protected Integer compute() { return ++i; }
        };
        assertThat(lazy.get(), equalTo(1));
        assertThat(lazy.get(), equalTo(1));
    }
}

File: engine-common/src/test/scala/com/sos/scheduler/engine/common/scalautil/LazyTest.scala
package com.sos.scheduler.engine.common.scalautil

import org.junit.runner.RunWith
import org.scalatest.FreeSpec
import org.scalatest.junit.JUnitRunner

/**
  * @author Joacim Zschimmer
  */
@RunWith(classOf[JUnitRunner])
final class LazyTest extends FreeSpec {

  "lazy" in {
    var i = 0
    val lazyI = Lazy { i += 1; i }
    assert(lazyI() == 1)
    assert(lazyI() == 1)
    assert(i == 1)
  }
}
